### ISSUE TITLE
Auto completion reveals es 6 js private variables which starting with #

### ISSUE THREAD

**Issue by subash1** [microsoft/monaco-editor#2169]
lets say a example 
```
class A {
      static #SOME  = 4;
      ......
}
```

if feed this class to editor for auto completion. 
if i type `A.`  it fine, SOME is not showing,
but if i create a object for A, like `var a = new A();`
then if i type `a.`, SOME is listing in auto completion.
any workaround to fix this.
Thanks

**Comment by spahnke** [comment]
Given the following playground code
```js
// validation settings
monaco.languages.typescript.javascriptDefaults.setDiagnosticsOptions({
    noSemanticValidation: false,
    noSyntaxValidation: false
});

// compiler options
monaco.languages.typescript.javascriptDefaults.setCompilerOptions({
    target: monaco.languages.typescript.ScriptTarget.ESNext,
    allowNonTsExtensions: true
});

var jsCode = `class A {
      static #SOME  = 4;
}

const a = new A();
a.`;

monaco.editor.create(document.getElementById('container'), {
    value: jsCode,
    language: 'javascript',
    // wordBasedSuggestions: false
});
```
does it look like this?
![grafik](https://user-images.githubusercontent.com/25029871/94684721-baa6c980-0328-11eb-8a2e-5d136ac14534.png)

If yes, then what you are experiencing are word based suggestions that are offered if no other suggestions were computed. The editor tries to be helpful and offers you suggestions based on words occurring in the document. You can disable those by setting `wordBasedSuggestions: false` (see commented line in the code above).

**Comment by subash1** [comment]
@spahnke , in my case that's show as class property.
![restrm](https://user-images.githubusercontent.com/6370601/94802461-5dcd1100-0405-11eb-80e2-12960fa79afe.png)

**Comment by spahnke** [comment]
That syntax is not fully standardized yet (see https://github.com/tc39/proposal-static-class-features). You can clearly see that in your example because the editor shows an error underneath the `#` character. So you would at least have to set the target to `ESNext`. But even then I'm getting an error in version 0.21.2 on the `static` keyword because TypeScript (which provides those suggestions) likely doesn't support this syntax yet, or I'm somehow missing some other settings that need to be made. Also for the future, it would be nice if you provided actual copyable code examples that can directly be used in the [playground](https://microsoft.github.io/monaco-editor/playground.html) and showcase your issue.
![grafik](https://user-images.githubusercontent.com/25029871/94806953-281d2d80-03ef-11eb-8e6e-3bbbc58c0c74.png)

To summarize: That syntax is likely not supported yet.

Example code used:
```js
// validation settings
monaco.languages.typescript.javascriptDefaults.setDiagnosticsOptions({
    noSemanticValidation: false,
    noSyntaxValidation: false
});

// compiler options
monaco.languages.typescript.javascriptDefaults.setCompilerOptions({
    target: monaco.languages.typescript.ScriptTarget.ESNext,
    allowNonTsExtensions: true
});

var jsCode = `class A {
      static #SOME  = 4;
}`;

monaco.editor.create(document.getElementById('container'), {
    value: jsCode,
    language: 'javascript',
    // wordBasedSuggestions: false
});
```

**Comment by hediet** [comment]
Closing due to inactivity.

If this bug still applies, please create a new issue. Make sure to select the form for bugs and include a minimal reproducible example. Thanks!
